### PR TITLE
Added .ngc support for sratools fasterqdump and prefetch

### PR DESCRIPTION
**Pull request by alexblaessle**
## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by apeltzer — APPROVED** [review]

File: modules/nf-core/sratools/fasterqdump/main.nf [hunk 1]
```diff
@@ -24,7 +24,17 @@ process SRATOOLS_FASTERQDUMP {
...
+    def key_file = ''
+
+    if (certificate){
+        if (certificate.toString().endsWith('.jwt')){
+            key_file += " --perm ${certificate}"
+            }
+        else if (certificate.toString().endsWith('.ngc')){
+            key_file += " --ngc ${certificate}"
+        }
+    }
```
Comment (Midnighter) [inline modules/nf-core/sratools/fasterqdump/main.nf:36]: You could reorganize this, but it's also fine to leave it as you have done.

```suggestion
        if (certificate.extension == 'jwt') {
            key_file += " --perm ${certificate}"
        }
        else if (certificate.extension == 'ngc') {
            key_file += " --ngc ${certificate}"
        }
```

**Review by Midnighter — APPROVED** [review]
Looks good. I think you can drop one level of conditionals, since a missing file will just end in the final `else` clause anyway, but it's certainly fine like this.

File: modules/nf-core/sratools/fasterqdump/main.nf (hunk 1 above)
Comment (alexblaessle) [reply modules/nf-core/sratools/fasterqdump/main.nf]: Thanks for the comment. Removed top level if statement in latest commit.

**Comment by apeltzer** [comment]
ping @maxulysse  to the rescue :D - this will be useful for https://github.com/nf-core/fetchngs/pull/238 too

**Comment by maxulysse** [comment]
ok, so as far as I can tell, the pair end tests are not working, but I'm fairly sure it's unrelated to your PR @alexblaessle, I'll have a look and try to come up with a separate PR to fix that so that we can merge your PR.
Expect a fix by the end of the day.

**Comment by maxulysse** [comment]
#4355 will solve the tests, sorry about that

**Comment by alexblaessle** [comment]
Thanks for all the help @apeltzer @maxulysse @Midnighter

**Pull request merged**